### PR TITLE
Fix extension not working on VS Code 10.12 due to a name change in the Node Debugger

### PR DESCRIPTION
**Pull request by digeff**
Fixes:
- https://github.com/Microsoft/vscode-react-native/issues/155
- https://github.com/Microsoft/vscode-react-native/issues/152

Tested on VS Code 10.11 and 10.12-Alpha
Screenshot of new version:
image

### CODE REVIEW

File: src/extension/rn-extension.ts [hunk 1]
```diff
@@ -70,7 +72,12 @@ export function deactivate(): void {
 }
 
 function configureNodeDebuggerLocation(): Q.Promise<void> {
-    const nodeDebugPath = vscode.extensions.getExtension("andreweinand.node-debug").extensionPath;
+    const nodeDebugExtension = vscode.extensions.getExtension("ms-vscode.node-debug") // We try to get the new version
+        || vscode.extensions.getExtension("andreweinand.node-debug"); // If it's not available, we try to get the old version
+    if (!nodeDebugExtension) {
+        return Q.reject<void>("Couldn't find the location of the node-debugger extension.");
```
Comment (hamiltonia) [inline src/extension/rn-extension.ts]: Should log this to give a hint why we to activate properly.

**Comment by hamiltonia** [comment]
:+1:

**Comment by MSLaguana** [comment]
Looks good to me

File: src/extension/rn-extension.ts [hunk 2]
```diff
@@ -1,9 +1,11 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for details.
 
-import {FileSystem} from "../common/node/fileSystem";
+import * as Q from "q";
```
Comment (meenakb) [inline src/extension/rn-extension.ts]: :+1:

File: src/extension/rn-extension.ts [hunk 3]
```diff
@@ -70,7 +72,12 @@ export function deactivate(): void {
 }
 
 function configureNodeDebuggerLocation(): Q.Promise<void> {
-    const nodeDebugPath = vscode.extensions.getExtension("andreweinand.node-debug").extensionPath;
+    const nodeDebugExtension = vscode.extensions.getExtension("ms-vscode.node-debug") // We try to get the new version
+        || vscode.extensions.getExtension("andreweinand.node-debug"); // If it's not available, we try to get the old version
+    if (!nodeDebugExtension) {
+        return Q.reject<void>(new Error("Couldn't find the location of the node-debugger extension."));
```
Comment (meenakb) [inline src/extension/rn-extension.ts]: Put the error in errorStrings.json and use ErrorHelper.GetError

Comment (digeff) [inline src/extension/rn-extension.ts]: Using ErrorHelper.getInternalError

**Comment by digeff** [comment]
@hamiltonia @MSLaguana Do you guys agree with these recent changes?

**Pull request merged**